### PR TITLE
TRUNK-5830: Migrate FieldType from Hibernate Mapping XML to JPA annotations

### PR DESCRIPTION
**Pull request by ManojLL**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5830

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by wikumChamith** [comment]
@ManojLL to fix the build error you might need to either exclude `date_changed` from the Hibernate mapping or add that column to the Liquibase schema. @dkayiwa, @ibacher  what would be the best solution here?

**Comment by dkayiwa** [comment]
Can you start by fixing the merge conflicts?

**Comment by ManojLL** [comment]
> Can you start by fixing the merge conflicts?

fixed the merge conflict

File: api/src/main/java/org/openmrs/FieldType.java [hunk 1]
```diff
@@ -9,20 +9,49 @@
...
+
 /**
  * FieldType
  */
+@Entity
+@Table(name = "field_type")
 @Audited
+@AttributeOverrides(value = {
+	@AttributeOverride(name = "name", column = @Column(name = "name", length = 50, nullable = false)),
+	@AttributeOverride(name = "retired", column = @Column(name = "retired", columnDefinition = "boolean default false"))
```
Comment (wikumChamith) [inline api/src/main/java/org/openmrs/FieldType.java:33]: Why use "boolean default false" here?'

Comment (ManojLL) [reply api/src/main/java/org/openmrs/FieldType.java:33]: we should set the `retired` column default value as `false` ?

Comment (wikumChamith) [reply api/src/main/java/org/openmrs/FieldType.java:33]: In BaseOpenmrsMetadata default value for retired is set to false.

https://github.com/openmrs/openmrs-core/blob/335c2b5186086f8a3b3d1d353bb1d06d23a20dcd/api/src/main/java/org/openmrs/BaseOpenmrsMetadata.java#L58-L60

Comment (ManojLL) [reply api/src/main/java/org/openmrs/FieldType.java:33]: without including `columnDefinition = "boolean default false"`, `HL7Service, OrderEntryIntergationTest, ` and `MedicationDispenseServiceTest` tests are getting failed.
 

Caused by:` org.h2.jdbc.JdbcSQLIntegrityConstraintViolationException: NULL not allowed for column "RETIRED"; SQL statement:
insert into FIELD_TYPE (FIELD_TYPE_ID, NAME, IS_SET, CREATOR, DATE_CREATED, UUID) values (?, ?, ?, ?, ?, ?) [23502-200]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:459)`

Comment (wikumChamith) [reply api/src/main/java/org/openmrs/FieldType.java:33]: @dkayiwa, @ibacher what are your thoughts on this?

Comment (dkayiwa) [reply api/src/main/java/org/openmrs/FieldType.java:33]: I do not like the column definition value of boolean default false
Can we think of an alternative?

**Pull request closed without merging**